### PR TITLE
Room settings, and enable encryption in unencrypted rooms (#212)

### PR DESCRIPTION
**Pull request by bmarty**
Only encryption status is displayed for the moment, other settings (room name, topic and avatar) will be available soon (waiting for design input)

### CODE REVIEW

File: vector/src/main/java/im/vector/riotx/features/roomprofile/settings/RoomSettingsAction.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.riotx.features.roomprofile.settings
+
+import im.vector.riotx.core.platform.VectorViewModelAction
+
+sealed class RoomSettingsAction : VectorViewModelAction {
+    data class SetRoomName(val newName: String) : RoomSettingsAction()
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/features/roomprofile/settings/RoomSettingsAction.kt]: Souldn't be nullable types?

File: vector/src/main/java/im/vector/riotx/features/roomprofile/settings/RoomSettingsViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    override fun handle(action: RoomSettingsAction) {
+        when (action) {
+            is RoomSettingsAction.EnableEncryption -> handleEnableEncryption()
+        }
+    }
+
+    private fun handleEnableEncryption() {
+        setState {
+            copy(currentRequest = Loading())
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/features/roomprofile/settings/RoomSettingsViewModel.kt]: Use ViewEvents?

File: vector/src/main/java/im/vector/riotx/core/epoxy/profiles/ProfileActionItem.kt [hunk 3]
```diff
@@ -43,11 +43,14 @@ abstract class ProfileActionItem : VectorEpoxyModel<ProfileActionItem.Holder>()
     @EpoxyAttribute
     var destructive: Boolean = false
     @EpoxyAttribute
-    lateinit var listener: View.OnClickListener
+    var listener: View.OnClickListener? = null
 
     override fun bind(holder: Holder) {
         super.bind(holder)
         holder.view.setOnClickListener(listener)
+        if (listener == null) {
```
[the remark this answers is not in the document]
Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/core/epoxy/profiles/ProfileActionItem.kt]: The `setOnClickListener` does it. Actually it does not fix what I wanted to fix (the `foreground` drawable ripple effect when it's not clickable...)

File: vector/src/main/java/im/vector/riotx/features/roomprofile/settings/RoomSettingsAction.kt (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/features/roomprofile/settings/RoomSettingsAction.kt]: Yes, maybe, this is not used for the moment, we will change it when implementing the feature

File: vector/src/main/java/im/vector/riotx/features/roomprofile/settings/RoomSettingsViewModel.kt (hunk 2 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/features/roomprofile/settings/RoomSettingsViewModel.kt]: We should have something out of the box (common parent for ViewEvents?) to manage this the same way in every screen

**Pull request merged**